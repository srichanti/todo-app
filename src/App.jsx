import React from 'react'
import { render } from 'react-dom'
import styled from 'styled-components'
import TodoList from './components/TodoList'
import TodoForm from './components/TodoForm'

const Container = styled.div`
    width: 250px;
    margin: 10px auto;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 13px
`

const App = () => (
    <Container>
        <TodoForm />
        <TodoList />
    </Container>
)

render(<App />, document.getElementById('app'))
