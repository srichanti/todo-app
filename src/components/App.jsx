import React from 'react'
import styled from 'styled-components'
import TodoList from './TodoList'
import TodoForm from './TodoForm'

const Container = styled.div`
  width: 250px;
  margin: 10px auto;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 13px;
`

const App = () => (
  <Container>
    <TodoForm />
    <TodoList />
  </Container>
)

export default App
