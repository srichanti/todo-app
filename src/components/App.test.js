import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe('App Component', () => {
  it('renders correctly', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})
