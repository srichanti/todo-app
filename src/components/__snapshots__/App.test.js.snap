// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TodoForm
          currentTodo=""
          handleOnchangeInput={[Function]}
          handleSubmit={[Function]}
        />,
        <TodoList
          todos={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentTodo": "",
          "handleOnchangeInput": [Function],
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "todos": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TodoForm
            currentTodo=""
            handleOnchangeInput={[Function]}
            handleSubmit={[Function]}
          />,
          <TodoList
            todos={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentTodo": "",
            "handleOnchangeInput": [Function],
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "todos": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
